### PR TITLE
Report modal

### PR DESCRIPTION
**Pull request by kuosmark**
# Description

- Fixes report modal.
- Adds input validation.
- Shows error notifications.
- Refactors icon sizing.

Fixes:
- [Report modal error handling and field validation](https://trello.com/c/wyHYeRc1/998-report-modal-error-handling-and-field-validation)
- [E2E tests for report feature](https://trello.com/c/vuZm3I0o/999-e2e-tests-for-report-feature)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unittest
- [x] Cypress e2e -tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/constants.ts [hunk 1]
```diff
@@ -34,7 +34,12 @@ export const palette = {
 };
 
 // Size
-export const DEFAULT_ICON_SIZE = { SMALL: 24, LARGE: 46 };
+export const DEFAULT_ICON_SIZE = {
+  SMALL: 23,
+  MEDIUM: 38,
+  LARGE: 40,
+  HUGE: 46,
+};
```
Comment (rottabonus) [inline src/components/constants.ts:42]: Its funny that we call it "default icon size" when we have 4 variants of it :D

I get that it has been expanded one by one, but maybe it is time to call it "icon_sizes" =D

File: cypress/tests/chat.cy.ts [hunk 2]
```diff
@@ -424,4 +396,57 @@ describe('chat', () => {
     cy.getByText('Estetyt keskustelut', 'a');
     cy.get('div[aria-label="unseen-messages-dot"]').should('not.exist');
   });
+
+  it('can report mentor succesfully', () => {
```
Comment (rottabonus) [inline cypress/tests/chat.cy.ts:400]: I would move these report -tests to "report.cy.ts"

**Review by rottabonus — COMMENTED** [review]
When the test pass, its good!

File: src/components/constants.ts (hunk 1 above)
Comment (kuosmark) [reply src/components/constants.ts:42]: True! 😃 I'll rename it!

File: cypress/tests/chat.cy.ts (hunk 2 above)
Comment (kuosmark) [reply cypress/tests/chat.cy.ts:400]: Yeah good suggestion!

File: src/components/constants.ts (hunk 1 above)
Comment (kuosmark) [reply src/components/constants.ts]: Done!

File: cypress/tests/chat.cy.ts (hunk 2 above)
Comment (kuosmark) [reply cypress/tests/chat.cy.ts]: Done!

**Pull request merged**